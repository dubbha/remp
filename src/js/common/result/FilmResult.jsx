import React from 'react';
import PropTypes from 'prop-types';
import './style.sass';

const Result = ({ film }) => (
  <div>
    {
      film.director
        ? `Films by ${film.director}`
        : 'Director unknown'
    }
  </div>
);

Result.propTypes = {
  film: PropTypes.shape({
    director: PropTypes.string,
  }).isRequired,
};

export default Result;
