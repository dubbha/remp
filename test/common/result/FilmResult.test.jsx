import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FilmResult from 'common/result/FilmResult';

describe('FilmResult', () => {
  let props;

  beforeEach(() => {
    props = {
      film: { director: 'QUENTIN TARANTINO' },
    };
  });

  it('should render successfully with director', () => {
    const wrapper = shallow(<FilmResult {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render successfully without director', () => {
    props.film.director = null;
    const wrapper = shallow(<FilmResult {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
