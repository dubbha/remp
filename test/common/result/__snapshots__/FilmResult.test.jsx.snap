// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmResult should render successfully with director 1`] = `
<div>
  Films by QUENTIN TARANTINO
</div>
`;

exports[`FilmResult should render successfully without director 1`] = `
<div>
  Director unknown
</div>
`;
